content/en/blog/2019/alb-ingress-gateway-iks/index.md

```diff
@@ -0,0 +1,394 @@
+---
+title: Securely direct traffic from IBM Cloud Kubernetes Service Ingress to Istio Ingress Gateway
+description: Configure the IBM Cloud Kubernetes Service Application Load Balancer to direct traffic to Istio Ingress gateway with mutual TLS.
+subtitle: Configure the IBM Cloud Kubernetes Service Application Load Balancer to direct traffic to Istio Ingress gateway with mutual TLS
+publishdate: 2019-11-07
+attribution: Vadim Eisenberg (IBM)
+keywords: [traffic-management,ingress,file-mount-credentials,iks]
+---
+
+In this blog post I show how to configure the [Ingress Application Load Balancer (ALB)](https://cloud.ibm.com/docs/containers?topic=containers-ingress-about)
```

Comment:
This overview section is packed with information, but I don't think it's doing a good job of explaining why users would want to set up this relatively complicated configuration. What is the goal here? If I'm understanding correctly, it is showing how to securely configure an application with a single entry point (the ALB domain name) where the services implementing the application are a mix of Istio and non-Istio services. Otherwise, if all the services are all Istio enabled, users would just use the gateway domain name, with no need for the ALB at all. Is that correct? I think it sort of eludes to that, but it doesn't say that clearly. If I'm understanding this correctly, I think even the title should be more along those lines (e.g., "Using an IKS ALB to securely access a partially Istio-enabled application".

Reply:
@frankbu 

> Otherwise, if all the services are all Istio enabled, users would just use the gateway domain name, with no need for the ALB at all.

Or the users still need some features of ALB and want to continue using it, even for Istio-enabled services. The blog post is intended for the users of IKS Ingress (ALB), they would know why they need to use ALB, which features they want.

Reply:
Right - users want to continue using the default IKS nginx based ALB because it gives them features via annotations that are not available with Istio Gateways. Chaining the ALB -> Istio Gateway -> Istio enabled apps allows them in theory to have the best of both worlds, but they need to know how to enable mTLS between the ALB pods and the Istio gateway pods